### PR TITLE
Add a new web account

### LINKED ISSUE #4: Add Username & Password for Web Account
Using the terminal, a user should be able to add a username and password associated with an account (like a website) so that they can view it later.

### PR DESCRIPTION
**Pull request by ryanloutos-wustl**

### CODE REVIEW

File: src/websiteAccount/WebsiteAccount.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import encryption.Encryption;
+import tables.WebsiteAccountEntity;
+import user.User;
+
+public class WebsiteAccount {
+	
+	private WebsiteAccountEntity websiteAccountEntity;
+	
+	public WebsiteAccount(User safeStoreUser, String nickname, String websiteLogin, String websitePassword) {
+			websiteAccountEntity = new WebsiteAccountEntity(safeStoreUser.getUserEntity(), Encryption.encrypt(nickname),
```
Comment (mbendy) [inline src/websiteAccount/WebsiteAccount.java]: just for coding guidelines, this is an extremely long line, maybe encrypt each string in a variable first to reduce the line length a little?

File: src/websiteAccount/WebsiteAccount.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	
+	public WebsiteAccount(User safeStoreUser, String nickname, String websiteLogin, String websitePassword) {
+			websiteAccountEntity = new WebsiteAccountEntity(safeStoreUser.getUserEntity(), Encryption.encrypt(nickname), 
+					Encryption.encrypt(websiteLogin), Encryption.encrypt(websitePassword));
+	}
+	
+	public WebsiteAccount(WebsiteAccountEntity websiteAccountEntity) {
+		this.websiteAccountEntity = websiteAccountEntity;
+	}
+
```
Comment (mbendy) [inline src/websiteAccount/WebsiteAccount.java:30]: can you add getters/setters for nickname, login, password similar to how i did mine in CreditCard?

File: test/queryWithDatabaseTests/WebsiteAccountTests.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			e.printStackTrace();
+		}
+	}
+
+	@Test
+	void testInsertWebsiteAccountToDatabase() {
+		try {
+			// insert both into database both ways
+			User testUser = new User(testUserEntity);
+			testUser.createSafeStoreAccount(databaseConnection);
```
Comment (mbendy) [inline test/queryWithDatabaseTests/WebsiteAccountTests.java]: don't use a method from User class, put the test user in the database directly using a dao, then be sure to delete it after your tests run. I'll push my tests so you can see how i did it

File: test/queryWithDatabaseTests/WebsiteAccountTests.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+			websiteAccountDao.create(testWebsiteAccountEntity);
+			
+			WebsiteAccount websiteAccountQuery = // insert query object into database
+					new WebsiteAccount(testUser, testAccountNickname, testAccountLogin, testAccountPassword);
+			websiteAccountQuery.addWebsiteAccount(databaseConnection);
+			
+			// pull from database both objects and make sure they are equal
+			WebsiteAccountEntity testEntityPulledFromDb = websiteAccountDao.queryForId(testWebsiteAccountEntity.getId());
```
Comment (mbendy) [inline test/queryWithDatabaseTests/WebsiteAccountTests.java]: this test is confusing, why are you querying twice? just query once, make sure the data is there.  Shouldn't need to delete two things? only one row should have been added?

**Review by mbendy — CHANGES_REQUESTED** [review]

File: src/websiteAccount/WebsiteAccount.java (hunk 1 above)
Comment (ryanloutos-wustl) [reply src/websiteAccount/WebsiteAccount.java]: line a bit short but still kind of long....think it should be good enough

File: src/websiteAccount/WebsiteAccount.java (hunk 2 above)
Comment (ryanloutos-wustl) [reply src/websiteAccount/WebsiteAccount.java:30]: done

File: test/queryWithDatabaseTests/WebsiteAccountTests.java (hunk 3 above)
Comment (ryanloutos-wustl) [reply test/queryWithDatabaseTests/WebsiteAccountTests.java]: fixed

File: test/queryWithDatabaseTests/WebsiteAccountTests.java (hunk 4 above)
Comment (ryanloutos-wustl) [reply test/queryWithDatabaseTests/WebsiteAccountTests.java]: you are right....fixed

**Review by mbendy — CHANGES_REQUESTED** [review]
Sorry, last thing, I forgot them too, but Javadocs :/

**Review by mbendy — APPROVED** [review]
looks good to me

**Pull request merged**